File: codechef/SPOTWO.py
def power(x,y):
	res=1
	while(y>0):
		if y%2==1:
			res=(res*x)%1000000007
		x=(x*x)%1000000007
		y=y/2
	return res%1000000007

def dec_to_bin(x):
    return int(bin(x)[2:])

t=input()
while t>0:
	n=(input())
	n=dec_to_bin(n)
	ans=power(2,n)
	ans=(ans*ans)%1000000007
	print ans
	t=t-1

File: codechef/SPOTWO.cpp
#include<cstdio>
#include<math.h>
#include<iostream>
using namespace std;
unsigned long long MOD = 1000000007ll;
unsigned long long power(unsigned long long  base, unsigned long long  exp) {
    unsigned long long res=1;
    while(exp>0) {
       if(exp%2==1) res=(res*base)%MOD;
       base=(base*base)%MOD;
       exp/=2;
    }
    return res%MOD;
}
int main()
{
   unsigned long long t;
   cin>>t;
   while(t--)
   {

        unsigned long long n,bin=0,arr[65]={0},pos=0,ans=0;
       cin>>n;
       int i;
        while(n>0)
        {
            arr[pos]=n%2;
            pos++;
            n=n/2;
        }
        //if(pos<=18)
        {
            for(i=pos-1;i>=0;i--)
            {
                bin=bin*10 + (arr[i]);
            }
            ans=power(2,bin);
            ans=(ans*ans)%MOD;
        }
        cout<<ans<<"\n";
   }
return 0;
}
